app/utils/legoAdapter/buildDeleteEntityReducer.ts

```diff
@@ -0,0 +1,38 @@
+import type { AsyncActionType } from 'app/types';
+import type { DeleteMeta } from 'app/utils/legoAdapter/asyncApiActions';
+import {
+  isAsyncApiActionBegin,
+  isAsyncApiActionFailure,
+  isAsyncApiActionSuccess,
+} from 'app/utils/legoAdapter/asyncApiActions';
+import type { EntityState } from '@reduxjs/toolkit/src/entities/models';
+import type { ActionReducerMapBuilder } from '@reduxjs/toolkit/src/mapBuilders';
+
+/**
+ * Handles optimistic and actual deletion of entities
+ */
+const buildDeleteEntityReducer = (
+  builder: ActionReducerMapBuilder<EntityState<unknown>>,
+  deleteActions: AsyncActionType[]
+) => {
+  builder.addMatcher(
+    isAsyncApiActionBegin.matching<DeleteMeta>(deleteActions),
+    (state, action) => {
+      state.ids = state.ids.filter((id) => id !== action.meta.id);
+    }
+  );
+  builder.addMatcher(
+    isAsyncApiActionFailure.matching<DeleteMeta>(deleteActions),
+    (state, action) => {
+      state.ids.push(action.meta.id);
+    }
+  );
```

Comment:
yeet

Reply:
Yeah, I agree optimistic delete is not really necessary. Probably better to simplify it